From 0ff278e1ae73060d9ab1304aec1a05432f446825 Mon Sep 17 00:00:00 2001
From: Paul Naoumenko <paulx.naoumenko@intel.com>
Date: Tue, 15 Oct 2013 17:55:55 +0200
Subject: [PORT FROM MAIN] wifi: make wpa_supplicant build possible for BOARD_WLAN_DEVICE other than bcmdhd

BZ: 139906

Make wpa_supplicant build possible for BOARD_WLAN_DEVICE other than bcmdhd.

Change-Id: Ifeae3f412cb2c9a16310cd865ae2fe6716182326
Orig-Change-Id: I07e08c744ac962c94ddcd2971507795664815234
Category: device enablement
Domain: CWS.WIFI-common
Origin: upstream
Upstream-Candidate: no,proprietary
Signed-off-by: Jeremie GARCIA <jeremie.garcia@intel.com>
---
 Android.mk |   21 +++++----------------
 1 files changed, 5 insertions(+), 16 deletions(-)

diff --git a/Android.mk b/Android.mk
index 46a101f..ff4ea7d 100644
--- a/Android.mk
+++ b/Android.mk
@@ -1,24 +1,13 @@
-LOCAL_PATH:= $(call my-dir)
+LOCAL_DIR:= $(call my-dir)
 
 ifndef WPA_SUPPLICANT_VERSION
 WPA_SUPPLICANT_VERSION := VER_0_8_X
 endif
-ifeq ($(WPA_SUPPLICANT_VERSION),VER_0_8_X)
-# The order of the 2 Android.mks does matter!
-# TODO: Clean up the Android.mks, reset all the temporary variables at the
-# end of each Android.mk, so that one Android.mk doesn't depend on variables
-# set up in the other Android.mk.
-include $(LOCAL_PATH)/hostapd/Android.mk \
-        $(LOCAL_PATH)/wpa_supplicant/Android.mk
-endif
-ifeq ($(WPA_SUPPLICANT_VERSION),VER_2_1_DEVEL)
-include $(call all-subdir-makefiles)
-endif
 
-ifeq ($(strip $(BOARD_WLAN_DEVICE)),wifi-bigcore)
-include $(LOCAL_DIR)/wpa_supplicant/Android.mk
-endif
 
-ifeq ($(strip $(BOARD_WLAN_DEVICE)),iwlwifi)
+ifeq ($(WPA_SUPPLICANT_VERSION),VER_0_8_X)
+ifeq (,$(filter iwlwifi wifi-bigcore,$(BOARD_WLAN_DEVICE)))
+include $(LOCAL_DIR)/hostapd/Android.mk
+endif
 include $(LOCAL_DIR)/wpa_supplicant/Android.mk
 endif
-- 
1.7.4.1

